### ISSUE TITLE
Recent update to --tag breaks backwards compatibility

### ISSUE THREAD

**Issue by johnmccabe** [openfaas/faas-cli#538]
## Expected Behaviour

It should be possible to pass the `--tag` flag to `build`/`push`/`deploy` and `generate` without specifying an option. The help text should also reflect the valid options that can be passed to this arg.

## Current Behaviour

The currently CLI sets a default value of `file` which is invalid and results in the following if used on its own (with `build` in this case):
```
$ faas build --tag -f myfn.yml
Please provide a valid --image name for your Docker image
```

## Possible Solution

PR incoming to fix the help text, and default values.

## Steps to Reproduce (for bugs)

1. Try to use the `--tag` flag on its own with no value.

## Context

Encountered this while pushing images to GCR

## Your Environment

* FaaS-CLI version ( Full output from: `faas-cli version` ):
```
CLI:
 commit:  dcddef2ae92880f50444103963408451abe82cc4
 version: 0.7.5
```

* Docker version ( Full output from: `docker version` ):
```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:31 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:29:02 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

* Are you using Docker Swarm (FaaS-swarm ) or Kubernetes (FaaS-netes)?
GKE

* Operating System and version (e.g. Linux, Windows, MacOS):
OSX

* Link to your project or a code example to reproduce issue:
n/a

**Comment by johnmccabe** [comment]
Invalid issue.

**Comment by alexellis** [comment]
Fixed in #539 

@johnmccabe please close this issue now and add it to the latest milestone using Derek.
